### PR TITLE
Add network-attached storage support

### PR DESCRIPTION
**Pull request by riidefi**

### CODE REVIEW

File: payload/sp/WideUtil.h [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    }
+}
+static inline void Util_toUtf8(char *dst, u32 dst_max, const wchar_t *src, u32 src_max) {
+    const u32 min_bound = MIN(dst_max, src_max);
+    const u32 len = wcslen(src /*, min_bound*/);
+
+    for (u32 i = 0; i < MIN(min_bound, len); ++i) {
+        dst[i] = (char)src[i];
+    }
+}
```
Comment (stblr) [inline payload/sp/WideUtil.h:21]: This is ASCII-only, is that intended?

File: payload/game/system/RootScene.c [hunk 2]
```diff
@@ -1,103 +1,8 @@
...
-}
 
 // Memory layout isn't randomized here
-void RootScene_allocateEx(RootScene *scn) {
-    RootScene_initNet(scn);
+void RootScene_allocateEx(RootScene * UNUSED(scn)) {
+    
 }
 
 PATCH_B(RootScene_allocate + 0xb64, RootScene_allocateEx);
```
Comment (stblr) [inline payload/game/system/RootScene.c:8]: This can be removed I think.

**Review by stblr — CHANGES_REQUESTED** [review]

File: payload/sp/WideUtil.h (hunk 1 above)
Comment (riidefi) [reply payload/sp/WideUtil.h:21]: Does the game ever read any non-ASCII paths? We could change the API to just forward UTF16 paths to the server.

**Comment by riidefi** [comment]
Resolved

File: payload/sp/WideUtil.h (hunk 1 above)
Comment (stblr) [reply payload/sp/WideUtil.h:21]: No, that would only be useful for ghost files which aren't supported with NetStorage.

**Review by stblr — APPROVED** [review]

**Pull request merged**